### PR TITLE
Docs: Data Sharing

### PR DESCRIPTION
**Pull request by erondpowell**
**Areas to Note:**

- [x] **What's a Share?** Checkout the `tips`.
- [x] **How it Works** copy somewhat unique, especially step 9. could be too casual @benhaynes 
- [x] **How it Works** Keep the `warning` or remove and wait fo bug fix? @rijkvanzanten 
- [x] **Limit Sharing by Collection, Limit Sharing by Item, Limit Sharing by Field**  
I have debated slashing these and just adding a hyperlink to the other docs. Any ideas?
- [x] **Extensibility Options** Still thinking about ideal boilerplate text for this and how to add in appropriate CTAs.

@rijkvanzanten Whenever you get around to it, this is pretty much ready for a review and merge.

### CODE REVIEW

**Review by benhaynes — CHANGES_REQUESTED** [review]
Looks really good! Just a few things I pointed out...

File: docs/app/data-sharing.md [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+1. Navigate to **Settings > Roles and Permissions**.
+2. Select the **[Role Name]** that is assigned to the Share.
+3. Find the **[Collection]** and click the icon in the column under the <span mi icon>visibility</span> icon and a small
+   menu will appear.
+4. Select <span mi icon>rule</span>**Use Custom** and a side menu will appear.
+5. Click **Field Permissions**.
+6. Click to check the Collection Fields you'd like to allow read permissions on.
+7. Select <span mi>check</span> in the side menu Header to confirm changes.
+
+## New User Options
```
[the remark this answers is not in the document]
Comment (erondpowell) [reply docs/app/data-sharing.md:201]: It is out of context... But it was an option available in **Data Share Options > Create Role > Users in Role** and Users, Roles, and Permissions was slightly under-developed. I will cut it and save it for Users, Roles, Permissions.

File: docs/app/data-sharing.md [hunk 2]
```diff
@@ -0,0 +1,246 @@
+# Data Sharing
+
+> Data Sharing allows you to share an Item from a Collection with other people that do not have the proper permissions
+> to view that Item.
```
[the remark this answers is not in the document]
Comment (erondpowell) [reply docs/app/data-sharing.md:4]: Incredible! 💪

**Comment by erondpowell** [comment]
@benhaynes @rijkvanzanten 
Take a look at this one guys. 
Seems done.

**Comment by erondpowell** [comment]
@rijkvanzanten Bumping this PR to your attention

File: docs/app/data-sharing.md [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+1. Navigate to **Settings > Roles and Permissions**.
+2. Select the **[Role Name]** that will be responsible for creating the Share.
+3. Find the **[Collection]** you wish to set Filter(s) for and click the icon in the column under
+   <span mi icon>share</span> and a small menu will appear.
+4. Select <span mi icon>rule</span>**Use Custom** and a side menu will appear.
+5. Create your Filter(s) as needed.
+
+## Limit Sharing by Field
+
+As mentioned several times before, when you create a Share on some Item, you associate a Role _(and thus the Role's
```
Comment (rijkvanzanten) [inline docs/app/data-sharing.md:186]:
```suggestion
When you create a Share on some Item, you associate a Role _(and thus the Role's
```

Lets avoid "As mentioned several times before" type of language across the docs as it has a bit of a negative connotation. It reads to me like "Like we told you many times before you dummy, ..."

File: docs/app/data-sharing.md [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+
+Working on an enterprise project and looking to outsource or financially sponsor the development of a Shares extension?
+Contact [our team](https://directus.io/contact/).
+
+:::
+
+## More Help
+
+Looking for technical support for your non-enterprise project? Chat with thousands of engineers within our growing
+[Community on Discord](https://discord.com/invite/directus).
```
Comment (rijkvanzanten) [inline docs/app/data-sharing.md:219]: Lets not include this in the page content. If we want to have a consistent footer for every page, it should be on the theme level, not the individual content of the pages.

Besides, for the context of Shares, extensions don't really come into play, so it doesn't have to be on this page regardless.

```suggestion
```

**Review by rijkvanzanten — CHANGES_REQUESTED** [review]

**Review by rijkvanzanten — APPROVED** [review]

**Pull request merged**